compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1293,7 +1293,19 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
         val formals1 =
           if (tp1.isVarArgsMethod && tp2.isVarArgsMethod) tp1.paramInfos.map(_.repeatedToSingle)
           else tp1.paramInfos
-        isApplicable(alt2, formals1, WildcardType) ||
+
+        class IsAsSpecific(methRef: TermRef, args: List[Type], resultType: Type)(implicit ctx: Context)
+          extends ApplicableToTypes(methRef, args, resultType) {
+          override def argOK(arg: TypedArg, formal: Type): Boolean =
+            // Simulate X* not being in subtyping relationship with any other type.
+            // This is necessary in cases when the last parameter of one method is repeated,
+            // and another is, for instance, Any - see neg/overload_repeated.scala.
+            // Note that we cannot just say that X* is really not a sub- or supertype of any other type,
+            // since that would interfere with typing method bodies.
+            !(arg.isRepeatedParam || formal.isRepeatedParam) && super.argOK(arg, formal)
```

Comment:
Repeated params should be comparable to other repeated params. So it should be
```scala
(arg.isRepeatedParam == formal.isRepeatedParam) && super.argOK(arg, formal)
```

Reply:
Here's a test case:
```
def foo(x: Any*)
def foo(x: String*)
foo("a")
```
Do we pick the 2nd? Why?

Reply:
`arg` and `formal` cannot both be repeated - in those case we drop the repeatedness in `isAsSpecific` before this test. I added a comment explaining as much.

This'd mean that in the test case we pick the 2nd, but we actually just fail the compilation since both overloads erase to the same signature.